compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/expr.rs

```diff
@@ -373,6 +377,21 @@ impl Expr {
         expr!(Array { elems }, typ)
     }
 
+    pub fn vector_expr(typ: Type, elems: Vec<Expr>) -> Self {
+        if let Type::Vector { size, typ: value_typ } = typ.clone() {
```

Comment:
Why is the `clone` needed here

Reply:
The use of `value_typ` later is a partial move out of the value.